src/node/desktop/src/native/desktop/desktop.cc

```diff
@@ -468,6 +468,67 @@ Napi::Value cleanClipboard(const Napi::CallbackInfo& info)
    return Napi::Value();
 }
 
+Napi::Value shortPathName(const Napi::CallbackInfo& info)
+{
+   Napi::String path = info[0].As<Napi::String>();
+
+#ifdef _WIN32
+
+   // Get path as UTF-8 text
+   std::string utf8Path = info[0].As<Napi::String>().Utf8Value();
+
+   // Figure out required size of string
+   std::wstring widePath(1024, 0);
+   {
+      int size = ::MultiByteToWideChar(
+         CP_UTF8, 0,
+         utf8Path.data(), utf8Path.size(),
+         &widePath[0], widePath.size()
+      );
+
+      if (size == 0) {
+         return path;
+      } else {
+         widePath.resize(size);
+      }
+   }
+
+   // Convert to a short path name
+   std::wstring wShortPath(1024, 0);
+   {
+      int size = ::GetShortPathNameW(widePath.data(), &wShortPath[0], wShortPath.size());
```

Comment:
Similar comment about 1024 being shorter than theoretically allowable path lengths. For complete robustness here we'd first want to query for the necessary size, adjust the buffer if necessary, then call it again.